src/app/models/message.ts

```diff
@@ -1,15 +1,15 @@
-import { Index } from './shared/index';
-
 export class Message {
   id: number;
   text: string;
-  group: Index;
-  session: Index;
+  groupId: number;
+  phaseId: number;
```

Comment:
Since phaseId will be unique no matter what session it's a part of, we'll be able to pull all messages for a phase just from that (instead of having to do a combination of sessionId + phaseId). This should make things easier with AngularFire2, since we can only query by one field at a time.